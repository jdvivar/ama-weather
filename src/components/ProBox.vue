<template>
  <div class="ProBox">
    <h3 v-if="title" class="title">{{ title }}</h3>
    <div v-if="image">
      <img :src="image" />
    </div>
    <div v-else class="well">
      <ul>
        <li v-for="(value, key) in data" :key="key">
          <div v-if="key === 'link'">
            <b>{{ key }}:</b>&nbsp;<a :href="value" target="_blank">See this
              info at Yahoo! Weather</a>
          </div>
          <div v-else>
            <div v-if="key !== 'code'">
              <b>{{key}}:</b> {{ value }}
            </div>
          </div>
        </li>
      </ul>
    </div>
  </div>
</template>

<script>
export default {
  name: 'pro-box',
  props: {
    title: String,
    image: String,
    data: Object
  }
}
</script>

<style lang="scss" scoped>
  .title {
    text-transform: capitalize;
  }

  .well {
    padding: 20px;
    background-color: rgba(black,.05);
    line-height: 1.42;
    font-family: monospace;
  }

  ul {
    list-style-type: none;
    padding: 0;
    margin: 0;
  }

  img {
    height: 80px;
  }
</style>
