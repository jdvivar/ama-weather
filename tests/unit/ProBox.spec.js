import { shallowMount } from '@vue/test-utils'
import ProBox from '@/components/ProBox'

describe('ProBox', () => {
  it('has correct markup', () => {
    const wrapper = shallowMount(ProBox)
    expect(wrapper.html()).toMatchSnapshot()
  })
})
