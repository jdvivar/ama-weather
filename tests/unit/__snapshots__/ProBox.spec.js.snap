// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProBox has correct markup 1`] = `
<div class="ProBox">
  <!---->
  <div class="well">
    <ul></ul>
  </div>
</div>
`;
